src/lxqtplatformtheme.cpp

```diff
@@ -222,8 +228,32 @@ bool LXQtPlatformTheme::usePlatformNativeDialog(DialogType type) const {
 QPlatformDialogHelper *LXQtPlatformTheme::createPlatformDialogHelper(DialogType type) const {
     if(type == FileDialog
        && qobject_cast<QApplication *>(QCoreApplication::instance())) { // QML may not have qApp
-        // use our own file dialog
-        return new LXQtFileDialogHelper();
+        // use our own file dialog provided by libfm
+
+        // When a process has this environment set, that means glib event loop integration is disabled.
+        // In this case, libfm-qt just won't work. So let's disable the file dialog helper and return nullptr.
+        if(qgetenv("QT_NO_GLIB") == "1") {
+            return nullptr;
```

Comment:
How about adding a message to let users know why the improved file dialog is unavailable?

Reply:
We can add a console log for it but I'm not sure if this is needed.

Reply:
console output will go unreaded to .xsession-errors - please don't